polygeist/tools/cgeist/Test/Verification/virt.cpp

```diff
@@ -46,24 +46,24 @@ void make() {
 // CHECK-NEXT:     %2 = llvm.getelementptr %arg0[0, 2] : (!llvm.ptr<struct<(struct<(i32)>, struct<(f32)>, f64)>>) -> !llvm.ptr<f64>
 // CHECK-NEXT:     llvm.store %arg2, %2 : !llvm.ptr<f64>
 // CHECK-NEXT:     %3 = llvm.mlir.addressof @str0 : !llvm.ptr<array<12 x i8>>
-// CHECK-NEXT:     %4 = llvm.getelementptr %3[0, 0] : (!llvm.ptr<array<12 x i8>>) -> !llvm.ptr<i8>
-// CHECK-NEXT:     call @_Z5printPc(%4) : (!llvm.ptr<i8>) -> ()
+// CHECK-NEXT:     %4 = "polygeist.pointer2memref"(%3) : (!llvm.ptr<array<12 x i8>>) -> memref<?xi8>
+// CHECK-NEXT:     call @_Z5printPc(%4) : (memref<?xi8>) -> ()
```

Comment:
Same: this cast looks odd to me

Reply:
See reply here https://github.com/intel/llvm/pull/7784/files#r1049943573